src/main/java/seedu/exercise/logic/commands/statistic/StatsFactory.java

```diff
@@ -0,0 +1,281 @@
+package seedu.exercise.logic.commands.statistic;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import javafx.collections.ObservableList;
+import seedu.exercise.model.ReadOnlyResourceBook;
+import seedu.exercise.model.property.Date;
+import seedu.exercise.model.property.Name;
+import seedu.exercise.model.property.Unit;
+import seedu.exercise.model.resource.Exercise;
+
+/**
+ * A class to generate Statistic depending on the chart type.
+ */
+public class StatsFactory {
+
+    private ObservableList<Exercise> exercises;
+    private String chart;
+    private String category;
+    private Date startDate;
+    private Date endDate;
+
+    /**
+     * Generates a StatsFactory object that can generate Statistic.
+     */
+    public StatsFactory(ReadOnlyResourceBook<Exercise> exercises, String chart, String category,
+                        Date startDate, Date endDate) {
+        this.exercises = exercises.getResourceList();
+        this.chart = chart;
+        this.category = category;
+        if (startDate == null && endDate == null) {
+            this.startDate = Date.getOneWeekBeforeToday();
+            this.endDate = Date.getToday();
+        } else {
+            this.startDate = startDate;
+            this.endDate = endDate;
+        }
+    }
+
+    /**
+     * Generates and returns statistic for different chart type.
+     */
+    public Statistic generateStatistic() {
+        HashMap<String, Double> data;
+        if (chart.equals("linechart")) {
+
+            return generateLineChartStatistic();
+
+        } else if (chart.equals("piechart")) {
+
+            return generatePieChartStatistic();
+
+        } else { //barchart
```

Comment:
you may want to add `else if (chart.equals("barchart"))` and a `else`  block that throws an exception

Reply:
done